syntax = "proto2";

package classification.protos;

message Spec {

    // classification model name.
    optional string model_name = 1;
    
    // Number of classes to predict.
    optional int32 num_classes = 2;

    // dataset directory which contains train and validation split.
    optional string dataset_dir = 3;
    
    // An offset for the labels in the dataset.
    optional int32 labels_offset = 4 [default=0];

    // Directory where train and eval logs are written to.
    optional string logdir = 5 [default="/tmp/logdir"];
    
    // The decay to use for the moving average. 
    // If left as None, then moving averages are not used.
    optional float moving_average_decay = 6;
}